homeassistant/components/media_player/__init__.py

```diff
@@ -654,23 +704,47 @@ def toggle(self):
         else:
             self.turn_off()
 
+    def async_toggle(self):
```

Comment:
Cool. Yeah, it seemed wrong for something so simple to go through two coroutines. I can't take credit for it myself, I found it on the web somewhere.